test/lockup/lockup-s-token.ts

```diff
@@ -0,0 +1,1447 @@
+				const [dev, property] = await init()
+				await dev.dev.approve(dev.lockup.address, 100)
+				const beforeBalance = await dev.dev
+					.balanceOf(deployer)
+					.then(toBigNumber)
+				const beforePropertyBalance = await dev.dev.balanceOf(property.address)
+				expect(beforeBalance.toString()).to.be.equal(deployerBalance.toString())
+				expect(beforePropertyBalance.toString()).to.be.equal('0')
+				await dev.lockup.depositToProperty(property.address, 100)
+				const afterBalance = await dev.dev.balanceOf(deployer).then(toBigNumber)
+				const afterPropertyBalance = await dev.dev.balanceOf(property.address)
+				expect(afterBalance.toString()).to.be.equal(
+					deployerBalance.minus(100).toString()
+				)
+				expect(afterPropertyBalance.toString()).to.be.equal('100')
+			})
+		})
+		describe('fail', () => {
+			it('Attempt to deposit money into an unauthenticated property.', async () => {
+				const [dev] = await init()
+				const propertyAddress = getPropertyAddress(
+					await dev.propertyFactory.create('test2', 'TEST2', user2, {
+						from: user2,
+					})
+				)
+				const res = await dev.lockup
+					.depositToProperty(propertyAddress, 100)
+					.catch(err)
+				validateErrorMessage(res, 'unable to stake to unauthenticated property')
+			})
+			it('0 dev staking is not possible.', async () => {
+				const [dev, property] = await init()
+				const res = await dev.lockup
+					.depositToProperty(property.address, 0)
+					.catch(err)
+				validateErrorMessage(res, 'illegal deposit amount')
+			})
+			it('user is not holding dev.', async () => {
+				const [dev, property] = await init()
+				const res = await dev.lockup
+					.depositToProperty(property.address, 100, { from: user3 })
+					.catch(err)
+				validateErrorMessage(res, 'ERC20: transfer amount exceeds balance')
+			})
+		})
+	})
+	describe('Lockup; deposit(update)', () => {
+		describe('success', () => {
+			it('update nft.', async () => {
+				const [dev, property, tokenId] = await init2()
+				const beforePosition = await dev.sTokenManager.positions(tokenId)
+				expect(beforePosition[0]).to.be.equal(property.address)
+				expect(beforePosition[1].toNumber()).to.be.equal(100)
+				expect(beforePosition[2].toNumber()).to.be.equal(0)
+				expect(beforePosition[3].toNumber()).to.be.equal(0)
+				expect(beforePosition[4].toNumber()).to.be.equal(0)
+				await dev.lockup.depositToPosition(tokenId, 100)
+				const afterPosition = await dev.sTokenManager.positions(tokenId)
+				expect(afterPosition[0]).to.be.equal(property.address)
+				expect(afterPosition[1].toNumber()).to.be.equal(200)
+				// TODO これでええんやろうか。。。なんか不安になてきた
+				expect(afterPosition[2].toString()).to.be.equal(
+					'100000000000000000000000000000000000'
+				)
+				// TODO これでええんやろうか。。。なんか不安になてきた
+				expect(afterPosition[3].toString()).to.be.equal('10000000000000000000')
+				// TODO これでええんやろうか。。。なんか不安になてきた
+				expect(afterPosition[4].toString()).to.be.equal('10000000000000000000')
```

Comment:
Check the following value with the old protocol:

```ts
const [dev, property] = await init()
await mine(1)
await dev.dev.deposit(property.address, 100)
await dev.dev.deposit(property.address, 100)
const value = await dev.lockup.getStoragePendingInterestWithdrawal(property.address, deployer)
// value
```

_In the first update, `cumlativeReward` and `pendingReward` are match._